snowflake_account_grant/main.tf.json

```diff
@@ -0,0 +1,48 @@
+{
+  "//": "NOTE: Generated by scripts/snowflake_generate_grant_all.\n\tChanges made directly to this file will be overwritten.\n\tMake improvements there so everyone can benefit.\n  The reason this module exists is that the provider only supports one grant resource per (database_name, schema_name, table_name, on_future, with_grant_option) tuple.\n\tFor example, if you used this module to grant an ALL privilege to a role you couldn't grant a subset of the ALL privs to another role.",
+  "variable": {
+    "per_privilege_grants": {
+      "type": "map(object({roles : list(string)}))",
+      "description": "A map of privileges to authorized roles and shares. Privileges must be UPPER case.\n  This allows you to authorize extra roles/shares for specific privileges.",
+      "default": {}
+    },
+    "roles": {
+      "type": "set(string)",
+      "description": "Grants privilege to these roles.",
+      "default": null
+    },
+    "with_grant_option": {
+      "type": "bool",
+      "description": "When this is set to true, allows the recipient role to grant the privileges to other roles.",
+      "default": null
+    }
+  },
+  "locals": {
+    "privileges": [
+      "CREATE DATABASE",
+      "CREATE INTEGRATION",
+      "CREATE ROLE",
+      "CREATE USER",
+      "CREATE WAREHOUSE",
+      "EXECUTE TASK",
+      "MANAGE GRANTS",
+      "MONITOR EXECUTION",
+      "MONITOR USAGE"
+    ]
+  },
+  "resource": {
+    "snowflake-account-grant-all": {
+      "all": {
```

Comment:
⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "snowflake-account-grant-all" in "required_providers"

Reply:
Seems like maybe something wrong with the code here?

Reply:
yeah, want to try it in an internal repo, learn, and write a better test

Reply:
something about the json syntax is probably messed up

Reply:
should it be called `snowflake_account_grant_all`?

Reply:
I'm now not sure if I trust these lint messages given https://github.com/chanzuckerberg/cztack/pull/262#discussion_r539423130

Reply:
>should it be called snowflake_account_grant_all?

that is also my preference (but I did the `-` because that seemed to be the style the majority of our team prefers)

Reply:
nvm, misinterpreted